### PR TITLE
fix(PeriphDrivers): Fix Pulldown/Pullup Configuration for ME30

### PR DESCRIPTION
**Pull request by mertekren**
This commit updates MXC_GPIO_Config according to GPIO hardware design. The register definitions shows similar padding functionality to MAX32675.

### Checklist Before Requesting Review

- [x] PR Title follows correct guidelines.
- [x] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

**Review by Jake-Carter — CHANGES_REQUESTED** [review]
Thanks @mertekren, just need to [format and lint](https://analogdevicesinc.github.io/msdk//CONTRIBUTING/#style-guide)

**Comment by Jake-Carter** [comment]
@sihyung-maxim should we merge this into main or `feat/ME30`?

**Comment by mertekren** [comment]
> Thanks @mertekren, just need to [format and lint](https://analogdevicesinc.github.io/msdk//CONTRIBUTING/#style-guide)
Done thanks

File: Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c [hunk 1]
```diff
@@ -100,165 +100,152 @@ int MXC_GPIO_Config(const mxc_gpio_cfg_t *cfg)
...
-/* ************************************************************************** */
-void MXC_GPIO_OutSet(mxc_gpio_regs_t *port, uint32_t mask)
-{
-    MXC_GPIO_RevA_OutSet((mxc_gpio_reva_regs_t *)port, mask);
-}
+    /* ************************************************************************** */
+    void MXC_GPIO_OutSet(mxc_gpio_regs_t * port, uint32_t mask)
+    {
+        MXC_GPIO_RevA_OutSet((mxc_gpio_reva_regs_t *)port, mask);
+    }
```
Comment (ozersa) [inline Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c:142]: After line 136 to end of file includes additional tab, need to be removed.

**Review by ozersa — CHANGES_REQUESTED** [review]
Additional tabs shall be removed.

**Comment by ozersa** [comment]
Target branch shall be feat/ME30 not main.
And description shall be cleared, remove template description.

**Comment by sihyung-maxim** [comment]
Agreed. Target branch should be `feat/ME30`.

File: Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c (hunk 1 above)
Comment (mertekren) [reply Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c]: Done, thanks. 
note: it did looked weird to me too. i guess they're introduced after i ran 'auto correction' via clang-format. anyways, removed them manually this time

**Comment by mertekren** [comment]
Note: i couldn't test this, there's no hardware for the test. (pulldown/pullup is not supported in fpga as I was informed)

**Review by Jake-Carter — APPROVED** [review]

File: Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c [hunk 2]
```diff
@@ -100,50 +100,36 @@ int MXC_GPIO_Config(const mxc_gpio_cfg_t *cfg)
...
+        default:
+            return E_BAD_PARAM;
+        }
+
+        // Configure the drive strength
+        if (cfg->func == MXC_GPIO_FUNC_IN) {
+            return E_NO_ERROR;
+        } else {
+            return MXC_GPIO_SetDriveStrength(gpio, cfg->drvstr, cfg->mask);
+        }
```
Comment (ozersa) [inline Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c:129]: This if else block should be out of switch case
image

File: Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c [hunk 3]
```diff
@@ -100,50 +100,36 @@ int MXC_GPIO_Config(const mxc_gpio_cfg_t *cfg)
...
     }
```
Comment (ozersa) [inline Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c:131]: Missing close bracket, build should be failed.
image

File: Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c [hunk 4]
```diff
@@ -100,50 +100,36 @@ int MXC_GPIO_Config(const mxc_gpio_cfg_t *cfg)
...
-    // Configure the drive strength
-    if (cfg->func == MXC_GPIO_FUNC_IN) {
-        return E_NO_ERROR;
-    } else {
-        return MXC_GPIO_SetDriveStrength(gpio, cfg->drvstr, cfg->mask);
-    }
-}
-
-/* ************************************************************************** */
+    /* ************************************************************************** */
```
Comment (ozersa) [inline Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c:132]: Please remove extra space

File: Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c [hunk 5]
```diff
@@ -198,7 +184,8 @@ void MXC_GPIO_DisableInt(mxc_gpio_regs_t *port, uint32_t mask)
 }
 
 /* ************************************************************************** */
-void MXC_GPIO_RegisterCallback(const mxc_gpio_cfg_t *cfg, mxc_gpio_callback_fn func, void *cbdata)
+void MXC_GPIO_RegisterCallback(const mxc_gpio_cfg_t *cfg, mxc_gpio_callback_fn func,
+                               void *cbdata)
```
Comment (ozersa) [inline Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c:188]: Please do not push unrelated change.

**Review by ozersa — CHANGES_REQUESTED** [review]
Comment added.

File: Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c [hunk 6]
```diff
@@ -100,50 +100,36 @@ int MXC_GPIO_Config(const mxc_gpio_cfg_t *cfg)
     }
 
     // Configure the pad
-    // TODO(ME30): "ps" (weak vs strong pull-up/down select) register field missing
     switch (cfg->pad) {
```
Comment (ozersa) [inline Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c:103]: Should be removed.
Duplicate line
image

Comment (mertekren) [reply Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c:103]: whoops done, thanks

File: Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c (hunk 3 above)
Comment (mertekren) [reply Libraries/PeriphDrivers/Source/GPIO/gpio_me30.c]: whoops done, thanks

**Review by ozersa — APPROVED** [review]
